Dockerfile

```diff
@@ -1,16 +1,15 @@
-FROM python:3.9.4@sha256:07c51c65ab9c1a156a1fb51eff3ec04feff7b85b2acb7d6cc65148b218d67402
+FROM python:3.12-bookworm@sha256:c50322f5f9b2a3b7ac68dc3cf03e5b29d7f51faa58b8321d975f028eb0c00a73
```

Comment:
`FROM python:3.12.1@sha256...` bookworm is not important as much as full version identifier.